server/src/weaverbird/pipeline/steps/text.py

```diff
@@ -1,13 +1,14 @@
+from datetime import datetime
 from typing import Literal
 
 from weaverbird.pipeline.steps.utils.base import BaseStep
 from weaverbird.pipeline.steps.utils.render_variables import StepWithVariablesMixin
 from weaverbird.pipeline.types import ColumnName
 
 
-class TextStep(BaseStep):
+class TextStep(BaseStep, smart_union=True):
     name: Literal["text"] = "text"
-    text: str
+    text: datetime | int | float | bool | str
```

Comment:
Some SQL backends might return `date` rather than `datetime`. I'm fine with handling only `datetime` here, but maybe we could add a conversion to the validator just to be 100% sure ? 

```python
if isinstance(value, date):
    value = datetime(date.year, date.month, date.day)
```

Reply:
the value stored in `text` comes from the interpolated jinja, it's not something "returned by the sql backend" :thinking:

Reply:
Woops yes true, I probably have `dtype` trauma :sweat_smile: